torch/autograd/gradcheck.py

```diff
@@ -790,7 +792,7 @@ def allclose_with_type_promotion(a, b, rtol, atol):
     return torch.allclose(a, b, rtol, atol)
 
 
-def vec_from_tensor(x, generator):
+def vec_from_tensor(x, generator, always_float64=False):
```

Comment:
nit: what you want here is not reall a "always_float64" but just always downcast complex to their element type.
Maybe a simple if statement here that transforms complex64 -> float32 and complex128 -> float64 would be enough?